collectors/apps.plugin/apps_plugin.c

```diff
@@ -1490,6 +1502,40 @@ static inline int read_proc_pid_stat(struct pid_stat *p, void *ptr) {
     return 0;
 }
 
+// ----------------------------------------------------------------------------
+
+static inline int get_pid_uptime(struct pid_stat *p) {
+    static procfile *global_uptime_ff = NULL;
```

Comment:
You shouldn't implement this again. There are 2 ways to get uptime, one from `clock_gettime()` and another one from `/proc/uptime`, and they have both been implemented in `collectors/proc.plugin`.

Reply:
Neither of those tell you how long a given process has been alive, just the system as a whole.

Reply:
The system uptime is what is set into the `global_uptime_ff` variable here though.

Reply:
This calculation takes place: `->uptime = (global_uptime > p->starttime)?(global_uptime - p->starttime):0;`

Reply:
So to clarify my initial comment, you should not implement getting **global** uptime again.

Reply:
I moved the functions for getting uptime to libnetdata.